src/api/eventTypes.js

```diff
@@ -100,23 +100,54 @@ export type PresenceEvent = $ReadOnly<{|
 |}>;
 
 /**
- * Updates the user status for a user
+ * New value for a user's self-assigned availability and/or text/emoji
+ *   status
  *
- * @prop [away] - update user's away status:
- *       - `true` the user is away regardless of presence
- *       - `false` remove the away status, now use presence
- * @prop [status_text] - if present:
- *       - empty string clears the user's status text
- *       - any string sets user's status to that
+ * See InitialDataUserStatus for the corresponding initial data. The
+ * per-user properties have different meanings between here and there.
  *
- * Not providing a property means 'leave this value unchanged'
  */
 export type UserStatusEvent = $ReadOnly<{|
   ...EventCommon,
   type: typeof EventTypes.user_status,
   user_id: UserId,
+
+  // Much of the below is observed empirically; we should get it into
+  // the doc:
+  //   https://chat.zulip.org/#narrow/stream/412-api-documentation/topic/Emoji.20statuses.20in.20zulip.2Eyaml/near/1322329
+
+  // true/false: User unavailable/available.
+  // omitted: User did not change this; keep current value
   away?: boolean,
+
+  // "foo": User's status text is "foo".
+  // "": User's status text is unset.
+  // omitted: User did not change this; keep current value
   status_text?: string,
+
+  // For these three emoji-status properties:
+  //
+  // non-empty string: User's status emoji is as indicated.
+  // "": User's status emoji is unset
+  // omitted: User did not change this; keep current value, or:
+  // TODO(server-5.0): Absent before feature level 86.
+  //
+  // "As indicated" means what the doc says, or:
```

Comment:
I don't quite understand this line. Is what follows meant to restate what's in the doc, or supplement it, or is it an alternative thing that the data could mean?

Reply:
I guess it was a too-terse way of saying, "If you're reading this after we've settled what actually belongs in the doc, then read the doc. Otherwise, read this, because we don't trust what the doc says now (e.g., it links to docs on reactions, which are unrelated)."